Player/Player.gd

```diff
@@ -181,6 +190,19 @@ func _process_movement(delta: float) -> void:
 	vel.x = hvel.x
 	vel.z = hvel.z
 	
+	if(vel.y < 0):
+		xVal = xVal
+	elif(vel.y > 0):
+		xVal = xVal
+	elif(vel.y == 0):
+		xVal = translation.y
```

Comment:
I think this is the relevant line. It could be moved inside the 
`if is_on_floor():
`
block on line 152.